src/Tribe/Views/V2/Messages.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * Handles a collection of View Messages.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2
+ */
+
+namespace Tribe\Events\Views\V2;
+
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class Messages
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2
+ */
+class Messages {
+	/**
+	 * A notice type of message.
+	 *
+	 * @since TBD
+	 */
+	const TYPE_NOTICE = 'notice';
+
+	/**
+	 * The strategy that will print a single message, the last, per priority collection, per type.
+	 *
+	 * @since TBD
+	 */
+	const RENDER_STRATEGY_PRIORITY_LAST = 'priority_last';
+
+	/**
+	 * The strategy that will print a single message, the first, per priority collection, per type.
+	 *
+	 * @since TBD
+	 */
+	const RENDER_STRATEGY_PRIORITY_FIRST =  'priority_first';
+
+	/**
+	 * An array of the messages handled by the object.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	protected $messages = [];
+
+	/**
+	 * The render strategy the collection will use to "render" the messages in the `to_array` method.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $render_strategy;
+
+	/**
+	 * Messages constructor.
+	 *
+	 * @param null|string $render_strategy The render strategy that should be used to render the messages in the
+	 *                                     `to_array` method.
+	 * @param array $messages A list of messages to hydrate the collection with.
+	 */
+	public function __construct( $render_strategy = null, array $messages = [] ) {
+		$this->render_strategy = $render_strategy ?: static::RENDER_STRATEGY_PRIORITY_LAST;
+		$this->messages        = $messages;
+	}
+
+	/**
+	 * Returns the human-readable message for a key.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key The message identification key or slug.
+	 * @param mixed $values A variadic number of arguments that should be used to fill in the message placeholders, if
+	 *                      the message contains `sprintf` compatible placeholders at all.
+	 *
+	 * @return string The human readable message for the specified key, if found, or the key itself.
+	 */
+	public static function for_key( string $key, ...$values ) {
+		$map = [
+			'no_results_found' => __( 'There were no results found.', 'the-events-calendar' ),
+			'month_no_results_found_w_keyword' => __(
+				'There were no results found for <strong>"%1$s"</strong> this month. Try searching next month.',
+				'the-events-calendar'
+			)
+		];
```

Comment:
:no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: 7aa1607edf1946dca81c6fd1c9182623</sub>